core/gesture.js

```diff
@@ -605,10 +695,34 @@ Blockly.Gesture.prototype.handleBlockStart = function(e, block) {
   this.mostRecentEvent_ = e;
 };
 
+/**
+ * Handle a mousedown/touchstart event on a bubble.
+ * @param {!Event} e A mouse down or touch start event.
+ * @param {!Blockly.Bubble} bubble The bubble the event hit.
+ * @package
+ */
+Blockly.Gesture.prototype.handleBubbleStart = function(e, bubble) {
+  goog.asserts.assert(!this.hasStarted_,
+      'Tried to call gesture.handleBubbleStart, but the gesture had already ' +
+      'been started.');
+  this.setStartBubble(bubble);
+  this.mostRecentEvent_ = e;
+};
+
 /* Begin functions defining what actions to take to execute clicks on each type
  * of target.  Any developer wanting to add behaviour on clicks should modify
  * only this code. */
 
+/**
+ * Execute a bubble click.
+ * @private
+ */
+Blockly.Gesture.prototype.doBubbleClick_ = function() {
+  // TODO (#1673): Consistent handling of single clicks.
```

Comment:
This comment should probably be updated to indicate that it's referencing a blockly issue